ppa/archive/forms.py

```diff
@@ -25,9 +25,31 @@ def valid_value(self, value):
 
 class SearchForm(forms.Form):
     '''Simple search form for digitized works.'''
+
+    SORT_CHOICES = [
+        ('relevance', 'Relevance'),
+        ('pub_date_asc', 'Year Newest-Oldest'),
+        ('pub_date_desc', 'Year Oldest-Newest'),
+        ('title_asc', 'Title A-Z'),
+        ('title_desc', 'Title Z-A'),
+        ('author_asc', 'Author A-Z'),
+        ('author_desc', 'Author Z-A'),
+    ]
+
+    def __init__(self, *args, **kwargs):
+        '''
+        Set choices dynamically based on form kwargs and presence of keywords.
+        '''
+        super(SearchForm, self).__init__(*args, **kwargs)
+        if not args or 'query' not in args[0] or not args[0]['query']:
+            # if there aren't keywords to search for, this will remove
+            # relevance from the form choices
+            self.fields['sort'].choices = self.fields['sort'].choices[1:]
```

Comment:
I tested manually submitting the form with sort set to relevance and no keyword and got the generic error message:
> Select a valid choice. relevance is not one of the available choices. 

If there's an easy way to do it, would be nice to customize the error message to add text that relevance is only available when search terms are used - but if it's not easy (and I can't think of an easy way off hand!), then we'll skip it for now.

Reply:
I bet I can do something in `form.clean()` on this. Let me think about a sane way to handle that scenario.